Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -1687,7 +1694,39 @@ private void LogCommandLineTooLong(SandboxedProcessInfo info)
                 allReportedFileAccesses: allFileAccesses,
                 detouringStatuses: result.DetouringStatuses,
                 maxDetoursHeapSize: maxDetoursHeapSize,
-                containerConfiguration: m_containerConfiguration);
+                containerConfiguration: m_containerConfiguration,
+                pipProperties: pipProperties);
+        }
+
+        private async Task UpdatePipPropertiesCountsAsync(SandboxedProcessResult result, Dictionary<string, int> pipProperties)
+        {
+            const string PipPropertyPrefix = "PipProperty_";
+
+            // Match everything between PipProperty_ and _EndProperty, assigned to a PipProperty_ group

```

Comment:
>PipProperty_ and _EndPropert [](start = 40, length = 28)

For testing this, here's what I think

- Create a separate struct/class and file for `PipPropertyFilter`. We can add description about `PipProperty_` and `_EndProperty` in that class.
```
class PipPropertyFilter
{
    private const string ErrorMessageGroup = "..";
    private const string NamedPropertyPrefix = "...";
    private const string NamedPropertySuffix = "...";
    private static readonly Regex s_pipNamedPropertyRegex = new Regex(NamedPropertyPrefix + @"(?<" + NamedPropertyPrefix + @">.+?)_NamedPropertySuffix ", RegexOptions.Compiled | RegexOptions.CultureInvariant | RegexOptions.Multiline);

    public readonly Regex Regex;
    public readonly string GroupName;
    private PipPropertyFilter(Regex regex, string groupName)
    {
       // ...
    }
   
    // Create a specialized error filter.
    public static PipPropertyFilter CreateErrorProperty(Regex regex) => new PipPropertyFilter(regex, ErrorMessageGroup);

    // Create a named filter.
    public static PipPropertyFilter CreateNamedProperty() => new PipPropertyFilter(s_pipNamedPropertyRegex, NamedPropertyPrefix);
}
```

- Create a separate file for `OutputFilter` struct. The `OutputFilter` struct now will take `PipPropertyFilter` as an argument.

- Create unit tests against the two classes.